test/Sema/accessibility_package_interface.swift

```diff
@@ -0,0 +1,38 @@
+// RUN: %empty-directory(%t)
+// RUN: split-file %s %t
+
+// RUN: %target-swift-frontend -emit-module %t/Utils.swift \
+// RUN:   -module-name Utils -swift-version 5 -I %t \
+// RUN:   -package-name swift-utils.log \
+// RUN:   -enable-library-evolution \
+// RUN:   -emit-module-path %t/Utils.swiftmodule \
+// RUN:   -emit-module-interface-path %t/Utils.swiftinterface \
+// RUN:   -emit-private-module-interface-path %t/Utils.private.swiftinterface
+
+// RUN: %target-swift-typecheck-module-from-interface(%t/Utils.swiftinterface) -I %t
+// RUN: %FileCheck %s --check-prefix=CHECK-PUBLIC < %t/Utils.swiftinterface
+// CHECK-PUBLIC: -module-name Utils
+// CHECK-PUBLIC-NOT: -package-name swift-utils.log
+// CHECK-PUBLIC: public func publicFunc()
+// CHECK-PUBLIC-NOT: package func packageFunc()
```

Comment:
I would recommend to list all the `CHECK-PUBLIC-NOT` check lines before the `CHECK-PUBLIC` lines. FileCheck looks for these in order. Here it would only error if a `package func packageFunc()` is after `public func publicFunc()`. Considering the order in the sources there could be a `packageFunc` printed before `publicFunc` and it wouldn't be reported.